### ISSUE TITLE
It is possible to draw outside of boundaries

### ISSUE THREAD

**Issue by madsane29** [4Q-s-r-o/signature#61]
It is not fixed. You still can draw outside of the boundaries to the right and to the bottom.

**Comment by madsane29** [comment]
I believe I fixed the problem. Pull Request here:

https://github.com/4Q-s-r-o/signature/pull/62

**Comment by crpo10** [comment]
its fixed or not? i have the same issue

**Comment by NitinSojitra** [comment]
I have the same issue.
User can draw right and bottom outside of signature pad.
Can you fix it or any solution for this issue?

**Comment by madsane29** [comment]
You can try out my fix if you add the following to your pubspec.yaml:

```
  signature:
    git:
      url: https://github.com/madsane29/signature
      ref: fix-draw-outside-of-canvas
```

Or if you download the modified signature.dart and add it to your project.

**Comment by madsane29** [comment]
Here is a small code to try it out:

```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:signature/signature.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Signature Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final SignatureController _controller = SignatureController();

  @override
  void initState() {
    super.initState();
    SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft]);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Signature(
          controller: _controller,
          width: MediaQuery.of(context).size.width * 0.7,
          height: MediaQuery.of(context).size.height * 0.7,
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: _controller.clear,
      ),
    );
  }
}

```

**Comment by NitinSojitra** [comment]
> You can try out my fix if you add the following to your pubspec.yaml:
> 
> ```
>   signature:
>     git:
>       url: https://github.com/madsane29/signature
>       ref: fix-draw-outside-of-canvas
> ```
> 
> Or if you download the modified signature.dart and add it to your project.

It works it only on bottom side and height and width doesn't work on overDraw.
User draws on right side.
Can you give me any other solution?

**Comment by NitinSojitra** [comment]
> Here is a small code to try it out:
> 
> ```
> import 'package:flutter/material.dart';
> import 'package:flutter/services.dart';
> import 'package:signature/signature.dart';
> 
> void main() {
>   runApp(const MyApp());
> }
> 
> class MyApp extends StatelessWidget {
>   const MyApp({Key? key}) : super(key: key);
> 
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       title: 'Signature Demo',
>       theme: ThemeData(
>         primarySwatch: Colors.blue,
>       ),
>       home: const MyHomePage(),
>     );
>   }
> }
> 
> class MyHomePage extends StatefulWidget {
>   const MyHomePage({Key? key}) : super(key: key);
> 
>   @override
>   State<MyHomePage> createState() => _MyHomePageState();
> }
> 
> class _MyHomePageState extends State<MyHomePage> {
>   final SignatureController _controller = SignatureController();
> 
>   @override
>   void initState() {
>     super.initState();
>     SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft]);
>   }
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       body: Center(
>         child: Signature(
>           controller: _controller,
>           width: MediaQuery.of(context).size.width * 0.7,
>           height: MediaQuery.of(context).size.height * 0.7,
>         ),
>       ),
>       floatingActionButton: FloatingActionButton(
>         child: Icon(Icons.add),
>         onPressed: _controller.clear,
>       ),
>     );
>   }
> }
> ```

it doesn't work.

**Comment by madsane29** [comment]
I don't know why it is not working for you. At me it is fine... 

![image](https://user-images.githubusercontent.com/40585482/178257054-31336afc-3973-4666-af6b-421fde48a0f7.png)

**Comment by NitinSojitra** [comment]
@madsane29 
Sorry bro, I wasted your time. 
I used MediaQuery in dialog box and I forget about full width.
When I put it in layout builder and set width using constraints.maxWidth.
It works.

Thank you so much

**Comment by MartinHlavna** [comment]
PR is merged and published in ```5.1.0```

**Comment by MartinHlavna** [comment]
Thanks!
